src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2800,30 +2800,33 @@ public override ImmutableArray<Diagnostic> GetMethodBodyDiagnostics(Cancellation
         /// </summary>
         public override ImmutableArray<Diagnostic> GetDiagnostics(CancellationToken cancellationToken = default)
         {
-            return GetDiagnostics(DefaultDiagnosticsStage, true, cancellationToken);
+            return GetDiagnostics(DefaultDiagnosticsStage, true, symbolFilter: null, cancellationToken);
         }
 
-        internal ImmutableArray<Diagnostic> GetDiagnostics(CompilationStage stage, bool includeEarlierStages, CancellationToken cancellationToken)
+        internal ImmutableArray<Diagnostic> GetDiagnostics(CompilationStage stage, bool includeEarlierStages, Predicate<ISymbolInternal>? symbolFilter, CancellationToken cancellationToken)
```

Comment:
>GetDiagnostics [](http://example.com/codeflow?start=44&length=14)

I think we should add a test that calls this helper with a `symbolFilter` and observes the effect of the filter. as well as observes the result of the next call, which doesn't pass any filter. I think we should target all unique filter points with a separate targeted test.